### DISCUSSION TITLE
Bundle's Product Parent Limitation

### DISCUSSION THREAD

**Discussion by jasonplumb** [Shopify/function-examples#320] [Feedback: Cart Transform API]
### Summary

Right now, when you create a bundle, you need to create a "product parent" to act as the bundle. There are two main points that feels non ideal:

1. The product parent does not work if set to draft mode meaning that the bundle product renders on the collection and product pages which is not ideal when the bundle is dynamically attached when the Shopify function runs
2. Creating a product parent is manual and does not appear to have a Shopify function approach to dynamically create the product parent.

While I'm interested to hear workarounds to the above, I would like to understand some of the reasons to treat the product parent as a product on Shopify? Would it be too complex to abstract the product parent and have a more automated approach during the shopify function runtime?

### The common use case

A customer wants to create a bundled product consisting of a range of drinks. Different drinks can form as part of the bundle and the user can add multiple combinations/bundles prior to checkout.

**Comment by ignacio-chiazzo** [comment]
Hi, thanks for the feedback. 

> The product parent does not work if set to draft mode meaning that the bundle product renders on the collection and product pages which is not ideal when the bundle is dynamically attached when the Shopify function runs

This is a very good point. We are considering supporting unavailable Product parents. Note that this will only be for the case of merge, more specifically: 
- Expand: The parent must be available. The components don't. 
- Merge: The components must be available; the parent doesn't. (Today, both must be available). 

> Creating a product parent is manual and does not appear to have a Shopify function approach to dynamically create the product parent.

Are you thinking about a custom product created on the fly? 

> While I'm interested to hear workarounds to the above, I would like to understand some of the reasons to treat the product parent as a product on Shopify? Would it be too complex to abstract the product parent and have a more automated approach during the shopify function runtime?

We want the bundle item to be modelled as a parent so that the merchant can keep track of the "product" that contained the list of items post-purchase. The parent attributes are stored as `LineItemGroups` after the order is placed, which is linked to the line items. 

Note that although we require a parent product, we are still considering supporting a custom Product created on the fly. 

> The common use case
> A customer wants to create a bundled product consisting of a range of drinks. Different drinks can form as part of the bundle and the user can add multiple combinations/bundles prior to checkout.

I would expect the merchant to create a product `Drinks Bundle` and the function to bundle the item.
